wakuv2/common/filter_test.go

```diff
@@ -0,0 +1,366 @@
+package common
+
+import (
+	crand "crypto/rand"
+	mrand "math/rand"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+	"golang.org/x/exp/maps"
+	"google.golang.org/protobuf/proto"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/waku-org/go-waku/waku/v2/payload"
+	"github.com/waku-org/go-waku/waku/v2/protocol"
+	"github.com/waku-org/go-waku/waku/v2/protocol/pb"
+)
+
+const testShard = "/waku/2/rs/16/32"
+
+type FilterTestCase struct {
+	f      *Filter
+	id     string
+	alive  bool
+	msgCnt int
+}
+
+func createLogger(t *testing.T) *zap.Logger {
+	config := zap.NewDevelopmentConfig()
+	config.Level = zap.NewAtomicLevelAt(zap.DebugLevel)
+	logger, err := config.Build()
+	require.NoError(t, err)
+	return logger
+}
+
+func generateFilter(t *testing.T, symmetric bool) (*Filter, error) {
+	var f Filter
+	f.Messages = NewMemoryMessageStore()
+
+	f.PubsubTopic = "test"
+
+	const topicNum = 8
+	f.ContentTopics = make(TopicSet, topicNum)
+	for i := 0; i < topicNum; i++ {
+		topic := make([]byte, 4)
+		_, err := crand.Read(topic) // nolint: gosec
+		require.NoError(t, err)
+		topic[0] = 0x01
+
+		f.ContentTopics[BytesToTopic(topic)] = struct{}{}
+	}
+
+	key, err := crypto.GenerateKey()
+	if err != nil {
+		t.Fatalf("generateFilter 1 failed")
+		return nil, err
+	}
```

Comment:
y not `require`? Is it for the error message? 🤔 
And I see it in more places below, not just here.
```suggestion
    require.NoError(t, err)
```

Reply:
Because of copy-pasting from another test :)  
Will fix it.